src/Orckestra.Composer.Search/Repositories/SearchRepository.cs

```diff
@@ -164,8 +164,16 @@ public virtual Task<ProductSearchResult> GetCategoryFacetCountsAsync(SearchCrite
             if (criteria.CultureInfo == null) { throw new ArgumentException(GetMessageOfNull(nameof(criteria.CultureInfo)), nameof(criteria)); }
             if (string.IsNullOrWhiteSpace(criteria.Scope)) { throw new ArgumentException(GetMessageOfNullWhiteSpace(nameof(criteria.Scope)), nameof(criteria)); }
 
-            var request = ProductRequestFactory.CreateProductRequest(criteria);
-            
+            SearchAvailableProductsBaseRequest request;
+            if (criteria is SearchBySkusCriteria searchBySkusCriteria)
+            {
+                request = ProductRequestFactory.CreateProductRequest(criteria);
+            }
+            else
+            {
+                request = ProductRequestFactory.CreateProductRequest(criteria.Scope);
+            }
```

Comment:
to understand better the logic let ry next, to see that it always like SearchAvailableProductsRequest:
 ```
var request = new SearchAvailableProductsRequest
            {
                Query = criteria is SearchBySkusCriteria ? CreateQuery(criteria ) : CreateQuery(criteria.scopeId)
            };
```